mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -259,11 +263,68 @@ def __init__(self, db_uri, default_artifact_root):
         if is_local_uri(default_artifact_root):
             mkdir(local_file_uri_to_path(default_artifact_root))
 
-        # Check if default experiment exists (not just if any experiments exist)
-        # This is important for databases that persist across test runs
+        self._initialize_store_state()
+
+    def supports_workspaces(self) -> bool:
+        return False
+
+    def _get_query(self, session, model):
+        """
+        Return a query for ``model``. Workspace-aware subclasses override this to enforce scoping.
+        """
+
+        return session.query(model)
+
+    @staticmethod
+    def _artifact_path_segments(uri: str | None) -> list[str]:
+        if not uri:
+            return []
+
+        if is_local_uri(uri):
+            # For filesystem artifact stores, use stdlib path handling to remain
+            # platform-safe (e.g., Windows drive letters and separators). Drive/root
+            # components are stripped for cross-platform segment comparison while
+            # leaving the original URI untouched.
+            path = local_file_uri_to_path(uri) if uri.startswith("file:") else uri
+            pure_path = PurePath(path)
+            drive_root = f"{pure_path.drive}{pure_path.root}"
+            skip_segments = {drive_root, pure_path.drive, pure_path.root, "", "/", "\\"}
+            return [segment for segment in pure_path.parts if segment not in skip_segments]
+
+        parsed = urlparse(uri)
+        path = parsed.path if parsed.scheme else uri
+        return [segment for segment in path.split("/") if segment]
+
+    def _add_fields(self, instance):
```

Comment:
```suggestion
    def _with_workspace_field(self, instance):
```
nit: since we don't plan to add more fields in near future, an explicit name would be easier to reason about.